tests/lib/test_azure.py

```diff
@@ -148,3 +158,93 @@ def token_provider() -> str:
 
     assert calls[0].request.headers.get("Authorization") == "Bearer first"
     assert calls[1].request.headers.get("Authorization") == "Bearer second"
+
+
+@pytest.mark.respx()
+def test_azure_api_key_redacted(respx_mock: MockRouter, _logger_with_filter: logging.Logger, caplog: pytest.LogCaptureFixture) -> None:
```

Comment:
note: I think you could've avoided the `_logger_with_filter` stuff by putting all these tests in a class and then doing
```py
@pytest.fixture(autouse=True)
def _logger_with_filter() -> logging.Logger: ...
```
but totally not worth blocking on that change

Reply:
Gotcha. Moved the tests in a class, however ruff still complains about the unused logger function, so kept it with underscore prefix.

Reply:
ah sorry tbc I meant that the class would just be for the logging tests as I assumed we wouldn't want the fixture to be setup for the other tests but if that would be fine then you don't need a class at all and it can just be at the module level

Reply:
Oh! I completely misunderstood, my bad 😆. Updating...